src/corecel/OpaqueId.hh

```diff
@@ -54,6 +54,14 @@ class OpaqueId
         return value_ != invalid_value();
     }
 
+    //! Increment ID's value
+    OpaqueId& operator++()
+    {
+        CELER_EXPECT(*this);
+        value_ += 1;
+        return *this;
+    }
```

Comment:
With the pointer analogy this is good. For completeness can you add a post-increment as well? (`operator++(int)`)? Can you also add to the `OpaqueId.test.cc` an example that uses `iota`?